x/bridge/types/msgs_test.go

```diff
@@ -0,0 +1,621 @@
+			expectedSigners: []sdk.AccAddress{addr1Bytes},
+			expectedValid:   true,
+		},
+		{
+			name: "invalid signer",
+			msg: types.MsgUpdateParams{
+				Sender: addr1,
+				NewParams: types.Params{
+					Signers:     []string{"qwerty"},
+					Assets:      []types.Asset{asset1, asset2},
+					VotesNeeded: types.DefaultVotesNeeded,
+					Fee:         math.LegacyNewDecWithPrec(5, 1),
+				},
+			},
+			expectedSigners: []sdk.AccAddress{addr1Bytes},
+			expectedValid:   false,
+		},
+		{
+			name: "duplicated signers",
+			msg: types.MsgUpdateParams{
+				Sender: addr1,
+				NewParams: types.Params{
+					Signers:     []string{addr2, addr2},
+					Assets:      []types.Asset{asset1, asset2},
+					VotesNeeded: types.DefaultVotesNeeded,
+					Fee:         math.LegacyNewDecWithPrec(5, 1),
+				},
+			},
+			expectedSigners: []sdk.AccAddress{addr1Bytes},
+			expectedValid:   false,
+		},
+		{
+			name: "empty assets",
+			msg: types.MsgUpdateParams{
+				Sender: addr1,
+				NewParams: types.Params{
+					Signers:     []string{addr1, addr2},
+					Assets:      []types.Asset{},
+					VotesNeeded: types.DefaultVotesNeeded,
+					Fee:         math.LegacyNewDecWithPrec(5, 1),
+				},
+			},
+			expectedSigners: []sdk.AccAddress{addr1Bytes},
+			expectedValid:   false,
+		},
+		{
+			name: "invalid asset",
+			msg: types.MsgUpdateParams{
+				Sender: addr1,
+				NewParams: types.Params{
+					Signers: []string{addr1, addr2},
+					Assets: []types.Asset{{
+						Id:       assetID1,
+						Status:   types.AssetStatus_ASSET_STATUS_UNSPECIFIED, // invalid status
+						Exponent: types.DefaultBitcoinExponent,
+					}},
+					VotesNeeded: types.DefaultVotesNeeded,
+					Fee:         math.LegacyNewDecWithPrec(5, 1),
+				},
+			},
+			expectedSigners: []sdk.AccAddress{addr1Bytes},
+			expectedValid:   false,
+		},
+		{
+			name: "duplicated assets",
+			msg: types.MsgUpdateParams{
+				Sender: addr1,
+				NewParams: types.Params{
+					Signers:     []string{addr1, addr2},
+					Assets:      []types.Asset{asset1, asset1},
+					VotesNeeded: types.DefaultVotesNeeded,
+					Fee:         math.LegacyNewDecWithPrec(5, 1),
+				},
+			},
+			expectedSigners: []sdk.AccAddress{addr1Bytes},
+			expectedValid:   false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			require.ElementsMatch(t, tc.msg.GetSigners(), tc.expectedSigners, "test: %v", tc.name)
+
+			if tc.expectedValid {
+				require.NoError(t, tc.msg.ValidateBasic(), "test: %v", tc.name)
+			} else {
+				require.Error(t, tc.msg.ValidateBasic(), "test: %v", tc.name)
+			}
+		})
+	}
+}
+
+// TestMsgChangeAssetStatus tests if MsgChangeAssetStatus messages are properly validated
+// and contain proper signers.
+func TestMsgChangeAssetStatus(t *testing.T) {
+	var (
+		pk1        = ed25519.GenPrivKey().PubKey()
+		addr1Bytes = sdk.AccAddress(pk1.Address())
+		addr1      = addr1Bytes.String()
+
+		assetID = types.AssetID{
+			SourceChain: "bitcoin",
+			Denom:       "btc",
+		}
+	)
```

Comment:
Do we need to keep redefining these every test or can we define once at the top level?